docs/introduction-to-smart-contracts.rst

```diff
@@ -208,7 +208,8 @@ The functions that make up the contract, and that users and contracts can call a
 The ``mint`` function sends an amount of newly created coins to another address.
 The :ref:`require <assert-and-require>` function call defines conditions that reverts all changes if not met.
 In this example, ``require(msg.sender == minter);`` ensures that only the creator of the contract can call ``mint``,
-and ``require(amount < 1e60);`` ensures a maximum amount of tokens. This ensures that there are no overflow errors in the future.
+and ``require(amount < 1e60);`` ensures a maximum amount of tokens: The function can be called very often,
+but due to gas costs, there is still a practical limit, and ``uint`` maximum value exceeds 1e77.
```

Comment:
I don't understand - how is `1e77` relevant here?

Reply:
Also, I think we should keep the comment about overflow, since this is the reason the restriction is there in the first place.

Reply:
Perhaps you've changed your mind about the overflow mention, since your prior suggestion "ensures a maximum amount of tokens: The function can be called very often, but due to gas costs, there is still a practical limit" removed it. Maybe in combination with my addition we could say "ensures a maximum amount of coins: The function can be called very often, but due to gas costs, there is still a practical limit, with overflow of ``uint amount`` only occurring in excess of 1e77 coins minted."

(I replaced the word "token" you used with "coins" here, as the document calls the unit of amount "coins" only.)

Providing the maximum value of uint is, as I mentioned in PR description, critical to understanding potential for overflow if it's to be discussed here. Overflow prevention in most programming doesn't focus on one operand, as was done here, and is maybe reasonable in crypto programming when it's not elsewhere. But although this doc mentions uints are "unsigned integer of 256 bits" earlier, astronomically large numbers are hard, and I think not many will go "oh right, that's WAY bigger than 1e60" unless stated right here, and it's easy enough to state.

Reply:
By not mentioning it, I did not mean to suggest it to be deleted, sorry.

You suggestion sounds good, but maybe we could also add the exact number of calls required for overflow?

Also it's probably better to split the first sentence into two, since what follows only concerns the second require.

Reply:
Wouldn't it be easier if we removed the line `require(amount < 1e60)`? The overflow check would prevent `balances[receiver] += amount;` from overflowing anyway. At least as an introduction, it's better to avoid this rather tricky idea that with the check `require(amount < 1e60)`, it would take more than `10**17` calls to `mint` to get an overflow (which wouldn't happen anyway because of safemath). And this is impractical because of gas reasons.

Reply:
@hrkrshnn Totally agree. It was my first thought opening this PR, but commit history showed a couple chances to have removed that code, so I took an improve before remove approach - but as @chriseth and I have walked through revisions, I think there is no good explanation simple enough for an intro. Were the line of code and comment about overflow not there at all I'd have breezed right through this part of the sample.

Reply:
Yep, sounds good! I would still prefer some documentation about potential overflows, though, since people should be aware.

Reply:
@chriseth I've updated the PR to remove instead. I've not read most of the rest of the docs, very new to Solidarity - looking around briefly, I see pretty clear warning about integers being restricted to ranges in docs/types.md, what happens on overflow. I also see the SMTChecker article which makes me think there's a way to shake out overflows before they happen if I'm taking this contract to production - that were I doing this production, might make me want to introduce a line or lines like `require(amount < 1e60)`.

Perhaps linking to types.md closer to initial discussion of uint in this intro would be sufficient?